src/pybind/mgr/dashboard/frontend/src/app/shared/services/crush.node.selection.service.ts

```diff
@@ -0,0 +1,142 @@
+import { AbstractControl } from '@angular/forms';
+
+import * as _ from 'lodash';
+
+import { CrushNode } from '../models/crush-node';
+
+export class CrushNodeSelectionService {
+  private nodes: CrushNode[] = [];
+  private easyNodes: { [id: number]: CrushNode } = {};
+  private allDevices: string[] = [];
+  private controls: {
+    root: AbstractControl;
+    failure: AbstractControl;
+    device: AbstractControl;
+  };
+
+  buckets: CrushNode[] = [];
+  failureDomains: { [type: string]: CrushNode[] } = {};
+  failureDomainKeys: string[] = [];
+  devices: string[] = [];
+  deviceCount = 0;
+
+  constructor() {}
+
+  initCrushNodeSelection(
+    nodes: CrushNode[],
+    rootControl: AbstractControl,
+    failureControl: AbstractControl,
+    deviceControl: AbstractControl
+  ) {
+    this.nodes = nodes;
+    nodes.forEach((node) => {
```

Comment:
Just got this error when I opened the ECP creation modal:
```
zone-evergreen.js:172 Uncaught TypeError: Cannot read property 'forEach' of undefined
    at ErasureCodeProfileFormModalComponent.initCrushNodeSelection (crush.node.selection.service.ts:32)
    at SafeSubscriber._next (erasure-code-profile-form-modal.component.ts:284)
    at SafeSubscriber.__tryOrUnsub (Subscriber.js:183)
```
![image](https://user-images.githubusercontent.com/399326/80091055-ffdff780-854f-11ea-8c7a-b3c4ae473295.png)

Plugin dropdown has a blank default and when I try to select the "Loading..." option all the validations stop.

Reply:
Still there after you reloaded the backend?

Reply:
No, this one was fixed by the reload.